### DISCUSSION TITLE
Rule on zbreceived

### DISCUSSION THREAD

**Discussion by gonzzzales** [arendst/Tasmota#13092] [Q&A] [answered]
Hello!
I am trying to set rule, to separate sensor values from one device to their own MQTT topic. Default it is like this
Rule1
  on zbreceived#0xD232#humidity do publish home/zigbee/office/humidity %value% endon
  on zbreceived#0xD232#temperature do publish home/zigbee/office/temperature %value% endon

But that work only, when SetOption100 is 0 and MQTT answer is {"**ZbReceived**":{"0xD232":{"Device":"0xD232","Temperature":25.82,"Humidity":98.04,"Pressure":999,"PressureScale":-1,"PressureScaledValue":9995,"SeaPressure":999,"Endpoint":1,"LinkQuality":189}}}
When i use SetOption100 1 i have MQTT messages like {"0xD232":{"Device":"0xD232","Temperature":28.41,"Humidity":97.6,"Pressure":999,"PressureScale":-1,"PressureScaledValue":9993,"SeaPressure":999,"Endpoint":1,"LinkQuality":199}} and rules not working.
Is it posssible to change Rule to work without ZbReceived?

**Comment by sfromis** [comment, marked as answer]
`on 0xD232#Humidity` should work, but I'd recommend using `ZbName` and `ZbNameKey` to assign friendly names instead of those short addresses.

As for getting separate topics, do take a look at `ZbDeviceTopic` and `ZbNameTopic`.

**Reply by gonzzzales** [reply]
Yes, work fine. Thank you!!!!
